pkg/sql/logictest/testdata/logic_test/truncate_with_concurrent_mutation

```diff
@@ -0,0 +1,346 @@
+# Make sure that table cannot be truncated if an index is being dropped
+# concurrently.
+statement ok
+CREATE TABLE t1(a int primary key, b int);
+
+statement ok
+CREATE INDEX idx_b ON t1(b);
+
+let $tbl_id
+SELECT id FROM system.namespace WHERE name = 't1';
+
+statement ok
+WITH to_json AS (
```

Comment:
doing manual mutation upsert instead of using pausing points because it takes too long for a test to run.

Reply:
🤔 if you set the cluster settings like this it should be faster: https://github.com/cockroachdb/cockroach/blob/724598b10f80fcf22065e57de066583df266bbbc/pkg/sql/logictest/testdata/logic_test/create_index#L252-L257

My sense is that it'd make for a better test.

Reply:
oh wow. that does help. will update it.